library/Icinga/Application/Hook/ApplicationStateHook.php

```diff
@@ -40,7 +40,7 @@ final public function getMessages()
     final public function addError($id, $timestamp, $message)
     {
         $id = trim($id);
-        $timestamp = (int) $timestamp;
+        $timestamp = (int) floor($timestamp);
```

Comment:
Please make *absolutely* sure we're dealing with floats here. My feeling is that this change isn't strictly necessary since we're dealing with timestamps and those are *usually* a result of `time()` or `DateTime->getTimestamp()`, in which case it's always an int.

There are probably other cases as well, I won't comment on each one. Please review them again and ensure on your own that you don't floor unnecessarily. It's maybe also an option to prevent decimals in the first place, given the calls are internal only. Though, an exception is `Benchmark`, as it already floored its timestamp previously so there's no problem to do it still.